src/lib/security/selinux_config.rb

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+
+module Security
+  # Class for handling SELinux kernel params
+  class SelinuxConfig
+    include Yast::Logger
+    Yast.import "Bootloader"
+
+    GETENFORCE_PATH = "/usr/sbin/getenforce".freeze
+    private_constant :GETENFORCE_PATH
+
+    class << self
+      DEFAULT_POLICY_OPTIONS = {
+        "security"  => :missing,
+        "selinux"   => :missing,
+        "enforcing" => :missing
+      }.freeze
+      private_constant :DEFAULT_POLICY_OPTIONS
+
+      def define_policy(name, options = {})
+        policies[name] = DEFAULT_POLICY_OPTIONS.merge(options)
+      end
+
+      def policies
+        @policies ||= {}
+      end
+
+      def policy_keys
+        DEFAULT_POLICY_OPTIONS.keys
+      end
+    end
+
+    define_policy(:disabled)
+    define_policy(:permissive, "security" => "selinux", "selinux" => "1", "enforcing" => :missing)
+    define_policy(:enforcing, "security" => "selinux", "selinux" => "1", "enforcing" => "1")
```

Comment:
to be honest I am not sure if this is the best way to define policies as it is spread over multiple places. Just consider if we want to add new policy where we need to add it. I prefer to have simple map of policies as private constant that explicitelly map policy with kernel options. E.g. here it confuse me why disabled does not have anything and why permissive has enforcing missing when it is default option. I am also not sure about that security => :missing as it can also mean that by disabling selinux you disable also if user define apparmor as security module.

Reply:
> I am also not sure about that security => :missing as it can also mean that by disabling selinux you disable also if user define apparmor as security module.

That's a good question. I'm not sure either. But I'm not sure even if at this point we can know if disabling SELinux could mean enabling AppArmor.

Reply:
> I prefer to have simple map of policies as private constant that explicitelly map policy with kernel options.

Actually I started with a [similar approach](https://github.com/yast/yast-security/pull/82/commits/5e9e3d120854b0a841d8fa96079d966974f0aa63) but I found this other way more _semantic_ and elegant. But let's _rollback_ to a _POLICIES_ map as you suggest.

Reply:
> But let's _rollback_ to a _POLICIES_ map as you suggest.

Ready. See https://github.com/yast/yast-security/pull/82/commits/9d11f6778f07e64d186df9d58beb787f79bcc743

Reply:
> But I'm not sure even if at this point we can know if disabling SELinux could mean enabling AppArmor.

I tried just removing the `selinux=1` to see if it is enough to disable SELinux, without touching `security=selinux` but it does not work. The system starts with SElLinux enabled in a permissive mode.

Again, I'm not sure if disabling SELinux should mean enabling AppArmor. I don't think so.

Reply:
Ok, I have digging a little bit more on the _major security LSM module_ and my conclusions are

* Unless I'm wrong, AppArmor is already enabled since kernel was compiled with [`CONFIG_DEFAULT_SECURITY_APPARMOR=y`](https://github.com/torvalds/linux/commit/2623c4fbe2ad1341ff2d1e12410d0afdae2490ca) and `CONFIG_LSM="integrity,apparmor"`, as it can be seen in the _/boot/config-\<kernel-version\>_ file (at least in openSUSE using a 5.10 kernel and SLES using a 5.3)
* [According to the documentation](https://www.kernel.org/doc/html/v5.6/admin-guide/LSM/apparmor.html#how-to-enable-disable), AppArmor will be disabled when using `apparmor=0` or `security=xxx` at boot time
* So, _deleting_ the `security=selinux` boot param looks safe since the user has to disable AppArmor on purpuose (either, with the `apparmor` param or using another major security module, like SELinux)

I did some tests and took the screenshots you can see below

| Not using `security` | `security=selinux` and `apparmor=1` | `security=apparmor` and `selinux=1`|
| - | - | - |
| ![no_params](https://user-images.githubusercontent.com/1691872/106463440-51c68e00-648f-11eb-8442-cdd75f46f3b3.png) | ![security_selinux_and_apparmor_1](https://user-images.githubusercontent.com/1691872/106463129-e7ade900-648e-11eb-8e71-f7ed0eacde8f.png) | ![security_apparmor_and_selinux_1](https://user-images.githubusercontent.com/1691872/106463193-04e2b780-648f-11eb-8fd0-24bb8aa8364d.png) |


Maybe we could have a more generic `LSMSecurityConfig` or `SecurityConfig` class to manage the security module to use (`security=xxx xxx=1`) and ask to its backend class for its configuration/policies/whatever. But I think it is out of the scope of this task.